File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands/game/BetTurnEndedCommand.java
package bluffinmuffin.protocol.commands.game;

import java.util.ArrayList;
import java.util.StringTokenizer;

import bluffinmuffin.poker.entities.type.GameRoundType;
import bluffinmuffin.protocol.commands.AbstractCommand;

public class BetTurnEndedCommand extends AbstractCommand
{
    @Override
    protected String getCommandName()
    {
        return BetTurnEndedCommand.COMMAND_NAME;
    }
    
    private final ArrayList<Integer> m_potsAmounts;
    private final GameRoundType m_round;
    
    public static String COMMAND_NAME = "gameBET_TURN_ENDED";
    
    public BetTurnEndedCommand(StringTokenizer argsToken)
    {
        final int count = Integer.parseInt(argsToken.nextToken());
        m_potsAmounts = new ArrayList<Integer>();
        for (int i = 0; i < count; ++i)
        {
            m_potsAmounts.add(Integer.parseInt(argsToken.nextToken()));
        }
        m_round = GameRoundType.values()[Integer.parseInt(argsToken.nextToken())];
    }
    
    public BetTurnEndedCommand(ArrayList<Integer> potsAmounts, GameRoundType round)
    {
        m_potsAmounts = potsAmounts;
        m_round = round;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        append(sb, m_potsAmounts.size());
        for (final int pa : m_potsAmounts)
        {
            append(sb, pa);
        }
        append(sb, m_round.ordinal());
    }
    
    public ArrayList<Integer> getPotsAmounts()
    {
        return m_potsAmounts;
    }
    
    public GameRoundType getRound()
    {
        return m_round;
    }
}

File: C#/PokerProtocol/Commands/Game/BetTurnEndedCommand.cs
using System.Collections.Generic;
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;
using PokerWorld.Game;
using PokerWorld.Game.Enums;

namespace PokerProtocol.Commands.Game
{
    public class BetTurnEndedCommand : AbstractCommand
    {
        public static string COMMAND_NAME = "gameBET_TURN_ENDED";

        private readonly RoundEnum m_Round;
        private readonly List<int> m_PotsAmounts;

        public RoundEnum Round
        {
            get { return m_Round; }
        }

        public List<int> PotsAmounts
        {
            get { return m_PotsAmounts; }
        } 

        public BetTurnEndedCommand(StringTokenizer argsToken)
        {
            int count = int.Parse(argsToken.NextToken());
            m_PotsAmounts = new List<int>();
            for (int i = 0; i < count; ++i)
            {
                m_PotsAmounts.Add(int.Parse(argsToken.NextToken()));
            }
            m_Round = (RoundEnum)int.Parse(argsToken.NextToken());
        }

        public BetTurnEndedCommand(List<int> potsAmounts, RoundEnum round)
        {
            m_PotsAmounts = potsAmounts;
            m_Round = round;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_PotsAmounts.Count);
            foreach( int pa in m_PotsAmounts )
                Append(sb, pa);
            Append(sb, (int)m_Round);
        }
    }
}
